### PR TITLE
For @ekr

### PR DESCRIPTION
**Pull request by martinthomson**
This doesn't address the suggested alternative design.

### CODE REVIEW

File: draft-ietf-webpush-vapid.md [hunk 1]
```diff
@@ -116,8 +116,8 @@ application server when choosing whether to discard a push message.
 ## Notational Conventions
 
 The words "MUST", "MUST NOT", "SHOULD", and "MAY" are used in this document.
-It's not shouting, when they are capitalized, they have the special meaning
-described in {{!RFC2119}}.
+When they are capitalized, they have the special meaning described in
+{{!RFC2119}}.
```
Comment (ekr) [inline draft-ietf-webpush-vapid.md:120]: Why not just cite RFC 8174

File: draft-ietf-webpush-vapid.md [hunk 2]
```diff
@@ -83,11 +83,11 @@ efficient or even sufficient.  Providing more information about the application
 server directly to a push service allows the push service to better distinguish
 between legitimate and bogus requests.
 
-Additionally, this design also relies on endpoint secrecy as any application
-server in possession of the endpoint is able to send messages, albeit without
-payloads.  In situations where usage of a subscription can be limited to a
-single application server, the ability to associate a subscription with the
-application server could reduce the impact of a data leak.
+Additionally, the design of RFC 8030 relies on maintaining the secrecy of push
```
Comment (beverloo) [inline draft-ietf-webpush-vapid.md:86]: Should RFC 8030 be a reference?

File: draft-ietf-webpush-vapid.md [hunk 3]
```diff
@@ -333,8 +333,8 @@ a public key.
 ## Using Restricted Subscriptions
 
 When a push subscription has been associated with an application server, the
-request for push message delivery MUST include proof of possession for the
-associated private key that was used when creating the push subscription.
+request for push message delivery MUST include a JWT signed by the private key
+that was used when creating the push subscription.
```
Comment (beverloo) [inline draft-ietf-webpush-vapid.md:337]: Is it correct to refer to the _private key_ that was used to create the subscription, when that is done using a _public key_ that was derived from the private key?

What about:

> When a push subscription has been **restricted to** an application server, the
> request for push message delivery MUST include a **valid** JWT signed by the
> private key **paired with the public key used when creating the push subscription**.

**Review by beverloo — APPROVED** [review]

**Comment by martinthomson** [comment]
Merged manually after resolving conflicts.

**Pull request closed without merging**